container/doctabs.go

```diff
@@ -0,0 +1,298 @@
+	"fyne.io/fyne/v2/theme"
+	"fyne.io/fyne/v2/widget"
+)
+
+// Declare conformity with Widget interface.
+var _ fyne.Widget = (*DocTabs)(nil)
+
+// DocTabs container is used to display various pieces of content identified by tabs.
+// The tabs contain text and/or an icon and allow the user to switch between the content specified in each TabItem.
+// Each item is represented by a button at the edge of the container.
+//
+// Since: 2.0
+type DocTabs struct {
+	baseTabs
+	CreateTab          func() *TabItem
+	OnClosed           func(*TabItem)
+	OnCloseIntercepted func(*TabItem)
+}
+
+// NewDocTabs creates a new tab container that allows the user to choose between various pieces of content.
+//
+// Since: 2.0
+func NewDocTabs(items ...*TabItem) *DocTabs {
+	tabs := &DocTabs{
+		baseTabs: baseTabs{
+			BaseWidget: widget.BaseWidget{},
+			current:    -1,
+		},
+	}
+	tabs.ExtendBaseWidget(tabs)
+	tabs.SetItems(items)
+	return tabs
+}
+
+// CreateRenderer is a private method to Fyne which links this widget to its renderer
+func (t *DocTabs) CreateRenderer() fyne.WidgetRenderer {
+	t.ExtendBaseWidget(t)
+	r := &docTabsRenderer{
+		baseTabsRenderer: baseTabsRenderer{
+			bar:         &fyne.Container{},
+			divider:     canvas.NewRectangle(theme.ShadowColor()),
+			indicator:   canvas.NewRectangle(theme.PrimaryColor()),
+			buttonCache: make(map[*TabItem]*tabButton),
+		},
+		docTabs:  t,
+		scroller: &Scroll{},
+	}
+	r.action = r.buildAllTabsButton()
+	r.create = r.buildCreateTabsButton()
+	r.box = fyne.NewContainer(r.create, r.action)
+	var lastX, lastY float32
+	r.scroller.OnScrolled = func(offset fyne.Position) {
+		// FIXME OnScrolled can be called when the offset hasn't changed (#1868)
+		if offset.X == lastX && offset.Y == lastY {
+			return
+		}
+		lastX = offset.X
+		lastY = offset.Y
+		r.updateIndicator(false)
+	}
+	r.updateAllTabs()
+	r.updateCreateTab()
+	r.updateTabs()
+	r.updateIndicator(false)
+	return r
+}
+
+// MinSize returns the size that this widget should not shrink below
+func (t *DocTabs) MinSize() fyne.Size {
+	t.ExtendBaseWidget(t)
+	return t.BaseWidget.MinSize()
+}
+
+// Declare conformity with WidgetRenderer interface.
+var _ fyne.WidgetRenderer = (*docTabsRenderer)(nil)
+
+type docTabsRenderer struct {
+	baseTabsRenderer
+	docTabs  *DocTabs
+	scroller *Scroll
+	box      *fyne.Container
+	create   *widget.Button
+}
+
+func (r *docTabsRenderer) Layout(size fyne.Size) {
+	r.updateAllTabs()
+	r.updateCreateTab()
+	r.updateTabs()
+	r.layout(&r.docTabs.baseTabs, size)
+	r.updateIndicator(true)
+}
+
+func (r *docTabsRenderer) MinSize() fyne.Size {
+	return r.minSize(&r.docTabs.baseTabs)
+}
+
+func (r *docTabsRenderer) Objects() []fyne.CanvasObject {
+	objects := r.baseTabsRenderer.Objects()
+	if i, is := r.docTabs.current, r.docTabs.Items; i >= 0 && i < len(is) {
+		objects = append(objects, is[i].Content)
+	}
+	return objects
+}
+
+func (r *docTabsRenderer) Refresh() {
+	// TODO Offset Scroller so current tab is always visible
```

Comment:
Agreed, I was hoping that the current tab would be scrolled into view

Reply:
Done